go/multipooler/manager/manager_test.go

```diff
@@ -965,3 +966,188 @@ func TestParseAndRedactPrimaryConnInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestActionLock_MutationMethodsTimeout(t *testing.T) {
+	ctx := context.Background()
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+
+	serviceID := &clustermetadatapb.ID{
+		Component: clustermetadatapb.ID_MULTIPOOLER,
+		Cell:      "zone1",
+		Name:      "test-service",
+	}
+
+	ts, _ := memorytopo.NewServerAndFactory(ctx, "zone1")
+	defer ts.Close()
+
+	poolerDir := t.TempDir()
+
+	// Create PRIMARY multipooler for testing
+	multipooler := &clustermetadatapb.MultiPooler{
+		Id:            serviceID,
+		Database:      "testdb",
+		Hostname:      "localhost",
+		PortMap:       map[string]int32{"grpc": 8080},
+		Type:          clustermetadatapb.PoolerType_PRIMARY,
+		ServingStatus: clustermetadatapb.PoolerServingStatus_SERVING,
+	}
+	require.NoError(t, ts.CreateMultiPooler(ctx, multipooler))
+
+	config := &Config{
+		TopoClient: ts,
+		ServiceID:  serviceID,
+		PoolerDir:  poolerDir,
+	}
+	manager := NewMultiPoolerManager(logger, config)
+	defer manager.Close()
+
+	// Start and wait for ready
+	senv := servenv.NewServEnv()
+	go manager.Start(senv)
+	require.Eventually(t, func() bool {
+		return manager.GetState() == ManagerStateReady
+	}, 5*time.Second, 100*time.Millisecond, "Manager should reach Ready state")
+
+	// Helper function to hold the lock for a duration
+	holdLock := func(duration time.Duration) context.CancelFunc {
+		lockCtx, cancel := context.WithCancel(context.Background())
+		lockAcquired := make(chan struct{})
+		go func() {
+			if err := manager.lock(lockCtx); err == nil {
+				// Signal that the lock was acquired
+				close(lockAcquired)
+				// Hold the lock for the duration or until cancelled
+				select {
+				case <-time.After(duration):
```

Comment:
Could change to `WithDeadline()` above and avoid the need for a separate `time.After()` case here?